cachix/src/Cachix/Client/Push.hs

```diff
@@ -20,6 +23,8 @@ where
 import qualified Cachix.Api as Api
 import Cachix.Api.Error
 import Cachix.Api.Signing (fingerprint, passthroughHashSink, passthroughHashSinkB16, passthroughSizeSink)
+import qualified Cachix.Client.Config as Config
+import Cachix.Client.Env (Env, config)
```

Comment:
There's a simple way to make an empty `Config` so maybe the pragmatic line could be to depend only on the that?

Reply:
`Config` certainly isn't as bad as `Env` if you can provide a `defaultConfig :: Config`. That way users only have to depend on the fields they need. `Env` otoh is quite a different beast with setup that a library user may not want, so switching to `Config` is an improvement. I'm not sure if Hercules CI will use `findPushSecret`, so it's more about separation of concerns.

Reply:
If you provide a `defaultConfig`, the `Maybe` doesn't seem necessary.

Reply:
Those are isomorphic and the user can use cli without the config and it's better to represent that case with `Nothing` rather than make defaultConfig implicit.

Reply:
That's absolutely a valid reason, but it's not a design concern for a library user who passes all arguments directly as before, which is why I'd like to maintain the separation of concerns and move `findPushSecret` into a module that's about the cli.

Reply:
I've added a TODO, currently all logic lives in `Cachix.Client.Push` so moving those modules around is work that shouldn't block this work.